### ISSUE TITLE
Move widgets with external dependencies to independent projects

### ISSUE THREAD

**Issue by piranna** [chjj/blessed#233]
Some widgets like Terminal has dependencies on other modules like pty.js and term.js. Instead of require the user to install them since they are somewhat heavyweight for normal cases, move them entirely to external modules and whoever that want to use them just need to use them directly as dependencies, without extra work.

**Comment by cancerberoSgx** [comment]
I agree 100% - Ideally, IMO, blessed should be as small as possible, focusing only on features like position, hierarchy, rendering, border, scrolling, events, focus, input and little more. The rest of features and high level components should be separate projects on top of it. So blessed project can focus only on portability and core issues. I think widgets, like image, video, terminal, table, etc should be moved to independent projects. Thanks
